libsolidity/interface/Version.cpp

```diff
@@ -44,36 +44,10 @@ string const dev::solidity::VersionStringStrict =
 	(string(SOL_VERSION_PRERELEASE).empty() ? "" : "-" + string(SOL_VERSION_PRERELEASE)) +
 	(string(SOL_VERSION_COMMIT).empty() ? "" : "+" + string(SOL_VERSION_COMMIT));
 
-bytes dev::solidity::binaryVersion()
-{
-	bytes ret{0};
-	size_t i = 0;
-	auto parseDecimal = [&]()
-	{
-		size_t ret = 0;
-		solAssert('0' <= VersionString[i] && VersionString[i] <= '9', "");
-		for (; i < VersionString.size() && '0' <= VersionString[i] && VersionString[i] <= '9'; ++i)
-			ret = ret * 10 + (VersionString[i] - '0');
-		return ret;
-	};
-	ret.push_back(uint8_t(parseDecimal()));
-	solAssert(i < VersionString.size() && VersionString[i] == '.', "");
-	++i;
-	ret.push_back(uint8_t(parseDecimal()));
-	solAssert(i < VersionString.size() && VersionString[i] == '.', "");
-	++i;
-	ret.push_back(uint8_t(parseDecimal()));
-	solAssert(i < VersionString.size() && (VersionString[i] == '-' || VersionString[i] == '+'), "");
-	++i;
-	size_t commitpos = VersionString.find("commit.");
-	solAssert(commitpos != string::npos, "");
-	i = commitpos + 7;
-	solAssert(i + 7 < VersionString.size(), "");
-	bytes commitHash = fromHex(VersionString.substr(i, 8));
-	solAssert(!commitHash.empty(), "");
-	ret += commitHash;
-	solAssert(ret.size() == 1 + 3 + 4, "");
-
-	return ret;
-}
+bytes const dev::solidity::VersionCompactBytes = {
+	ETH_PROJECT_VERSION_MAJOR,
```

Comment:
I don't see a reason for decoding the version string here at runtime instead of passing in the version as numbers at compile time - this is much simpler and has the additional advantage that compilation should fail, if we'd happen to have version numbers >=256.